notebook.py

```diff
@@ -20,40 +24,571 @@ def _():
+                            "Level 5",
+                            "ERROR",
+                        ),
+                    ),
+                ),
+            ),
+        )
+        return dataframe
+
+    def mime_types(dataframe: pd.DataFrame) -> pd.DataFrame:
+        """Add mime type based on file extension."""
+        mimetypes.add_type("application/vnd.ms-outlook", ".msg")
+        dataframe.loc[:, "mimetype"] = dataframe["extension"].apply(
+            lambda extension: (
+                "unknown"
+                if pd.isna(extension) or not extension
+                else mimetypes.types_map.get(extension, "unknown")
+            )
+        )
+        return dataframe
+
+    def is_digitized_aip(dataframe: pd.DataFrame) -> pd.DataFrame:
+        """Identifies digitized AIPS based on UUID."""
+        digitized_aip_regex = r"\d{4}_\d{3}[r|R]{2}_\d{3}"
+        dataframe.loc[:, "is_digitized_AIP"] = np.where(
+            dataframe.accession_name.str.contains(digitized_aip_regex, regex=True),
+            "Digitized",
+            np.where(
+                dataframe.accession_name.isin(os.environ["DIGITIZED_BAG_IDS"].split(",")),
+                "Digitized",
+                "Born Digital",
+            ),
+        )
+        return dataframe
+
+    def is_replica(dataframe: pd.DataFrame) -> pd.DataFrame:
+        """Identifies replicas based on S3 bucket."""
+        dataframe.loc[:, "is_replica"] = np.where(
+            dataframe["bucket"].str.contains("4b"),
+            True,
+            np.where(dataframe["bucket"].str.contains("5b"), True, False),
+        )
+        return dataframe
+
+    def is_normalized_file(dataframe: pd.DataFrame) -> pd.DataFrame:
+        """Identifies normalized files based on several criteria."""
+        am_uuid_regex = (
+            r"-\S{8}-\S{4}-\S{4}-\S{4}-\S{12}."  # regex for archivematica file UUID
+        )
+        dataframe.loc[:, "is_normalized_file"] = np.where(
+            dataframe.file.str.contains(am_uuid_regex, regex=True),
+            True,
+            np.where(
+                dataframe.file.str.contains("data/thumbnails"),
+                True,
+                np.where(
+                    (dataframe["is_digitized_AIP"] == "Digitized")
+                    & (dataframe.file.str.contains(".pdf")),
+                    True,
+                    False,
+                ),
+            ),
+        )
+        return dataframe
+
+    def set_status(dataframe: pd.DataFrame) -> pd.DataFrame:
+        """Adds a status based on CDPS content categories."""
+        dataframe.loc[:, "status"] = np.where(
+            dataframe["is_replica"],
+            "replica",
+            np.where(
+                dataframe["is_normalized_file"],
+                "normalized/access",
+                np.where(dataframe["is_metadata"], "metadata", "original content"),
+            ),
+        )
+        return dataframe
+
```

Comment:
The business logic of these functions is largely copied over from Charlie's Jupyter notebook